server/connector.go

```diff
@@ -6,25 +6,30 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"github.com/google/uuid"
 	"io"
 	"net"
 	"sync"
 	"sync/atomic"
 	"time"
 
+	"github.com/google/uuid"
+
 	"golang.ngrok.com/ngrok"
 	"golang.ngrok.com/ngrok/config"
 
 	"github.com/go-kit/kit/metrics"
 	"github.com/itzg/mc-router/mcproto"
 	"github.com/juju/ratelimit"
 	"github.com/pires/go-proxyproto"
+	"github.com/sethvargo/go-retry"
 	"github.com/sirupsen/logrus"
 )
 
 const (
-	handshakeTimeout = 5 * time.Second
+	handshakeTimeout     = 5 * time.Second
+	backendTimeout       = 30 * time.Second
```

Comment:
Can  you double check this is less than the timeout that the Minecraft client seems to run when attempting to connect?